src/main/java/eu/darkbot/popcorn/def/CaptchaPicker.java

```diff
@@ -104,13 +68,105 @@ public boolean canRefresh() {
 
     @Override
     public String status() {
-        return "Solving captcha" + (currentMatch != null ? " " + currentMatch : "");
+        return "Solving captcha" + (boxMatch != null ? " " + boxMatch : "");
     }
 
     @Override
     public void tick() {
+        boxes.stream()
+                .filter(this::findBox)
+                .min(Comparator.comparingDouble(box -> hero.locationInfo.now.distance(box)))
+                .ifPresent(box -> {
+                    if (!boxMatch.hasAmount && amountToCollect == 0) {
+                        amountToCollect = (int) boxes.stream()
+                                .filter(b -> b.type.equals(boxMatch.name))
+                                .count();
+                    } else if (boxMatch.hasAmount && amountToCollect == 0) {
+                        amountToCollect = boxMatch.amount;
+                    }
+                    if (isNotWaiting()) collectBox(box);
+                });
```

Comment:
This is a bit of aconvoluted mess.
Simply find what the next box to pick should be (if any) and then pick that up, don't do that much logic inside an if present